### DISCUSSION TITLE
Zksync endless deploy loading

### DISCUSSION THREAD

**Discussion by Yanabonne** [Cyfrin/foundry-full-course-cu#2114] [Q&A] [answered]
<img width="275" alt="image" src="https://github.com/user-attachments/assets/eeeefa93-a8a7-4146-b34a-cbb8cd971a39">

When I try to deploy my code on zksync using wallet, it starts loading and never stops. I tried reloading everything, changing zksolc, but nothing works and there are no errors in the terminal. It got deployed once a couple days ago somehow, but other than that I cannot deploy anything on zksync. Any help is appreciated

**Comment by EngrPips** [comment, marked as answer]
Hello @Yanabonne, I will say you should check your internet connection.

**Reply by Yanabonne** [reply]
@EngrPips thank you for your answer. If my internet connection is not stable enough, why all other functions and deployments are working fine.. I was just curious to see if anyone else had the same issue. But probably you are correct

**Reply by EngrPips** [reply]
Yeah, I understand it can be confusing sometimes. You might as well try another browser.

**Comment by Vishalll07** [comment]
hey @EngrPips  Hii remember me ,proly not , ok so could u please lemme know how much time it takes to verify , firstly it wasn't even deploying but after refreshing the page it deployed in the remote devnet environment of the zksync plugin , now after selecting wallet , hitting verify and deploy nd after clicking sign in metamask  its not giving a success message in the console as you can see below
![eth8](https://github.com/user-attachments/assets/db23d7d0-a441-4396-b0c6-ae3ba32bee77)

**Reply by EngrPips** [reply]
Hey @Vishalll07, I think I remember you. I think the `verify and deploy` feature is not working properly as of now.

**Reply by Vishalll07** [reply]
ohhk got it !! then i guess i will be unable to look at my code in `zksync sepolia explorer`

**Reply by EngrPips** [reply]
Working with `zksync` isn't as very smooth as working with `Ethereum` at the moment, but that will change over time.

**Reply by Vishalll07** [reply]
ok ok got it, thanks man

**Comment by phelipebf** [comment]
I figured out it's a problem with contract verify feature. Just uncheck "Verify Contract" checkbox and everything seems to work.

**Reply by EngrPips** [reply]
Interesting, Thanks for the heads-up.

**Reply by tauren** [reply]
How long does it take before a response shows in the terminal? I had network wrong the first deploy. The next two deploys I had *verify* checked and let it sit for hours. Then I tried deploying without verify, and its doing the same thing. Still waiting after 10 minutes. I'm getting the feeling something is broken with the zksync or its plugin as I was able to deploy to Sepolia testnet just fine. 

<img width="1114" alt="image" src="https://github.com/user-attachments/assets/6c4eb4b6-8f5d-48ec-b4ee-ac3ce900d386">

**Reply by EngrPips** [reply]
Hello @tauren, refresh the browser, then retry. Let's see if that will help

**Reply by tauren** [reply]
> Hello @tauren, refresh the browser, then retry. Let's see if that will help

That solved it! How embarrassing! 🤦

**Reply by EngrPips** [reply]
Programming is indeed fun.
